_RFCs/0002-scaffolding-rfc.md

```diff
@@ -0,0 +1,56 @@
+
+That said - an experienced user of Habitat should be able to use and build upon a scaffolding to package and run more complex applications. Scaffoldings should allow customization through overriding callbacks, environmental variables, plan settings, helper functions, and more. However, this kind of customization should not be required for simple applications - it should be an option, not a requirement.
+
+Additionally, scaffoldings should not require the use of alpha features - features should be out of alpha and beta to be required by scaffoldings. Although users are welcome to experiment with combining alpha features and scaffolding, they are not officially supported by the core-plans maintainers.
+
+# Reference-level explanation
+
+Habitat scaffoldings were originally modeled after [Heroku Buildpacks](https://devcenter.heroku.com/articles/buildpacks). They provide both a way of packaging an application and an executable runtime. If an advanced user prefers to use only the packaged application they can override or ignore the runtime functionality - but the default functionality will always include a runtime.
+
+A user of scaffolding should be able to package and run a simple application
+* with minimal Habitat knowledge
+* within 1 hour
+* with minimal operations experience
+
+Additionally, using a scaffolding should not depend on alpha or beta features (which at the time of this RFC includes composite plans). It is fine for someone to experiment with using scaffoldings with alpha/beta features, but this should not be required for use and should not be officially supported by the core maintainers of core-plans.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This will tip the appeal of scaffoldings toward beginner users, rather than advanced users. It will take additional work to establish documentation and patterns of more advanced uses of scaffolding. We will also need to communicate clearly what Habitat features we do and don't officially support for use with scaffoldings.
+
+This may also require re-working of some of our scaffoldings.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+I had originally [proposed splitting the node scaffolding](https://github.com/habitat-sh/core-plans/pull/1009). Through vigorous debate and discussion, we determined that splitting may not be necessary - clarifying of the purpose and intended use of scaffolding was. These guidelines should apply to all scaffoldings, not just the node scaffolding.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- How will we and who will update the existing scaffoldings to these standards?
```

Comment:
Most of this feels pretty reasonable to me. I wondered if you could share which of the currently released scaffoldings don't adhere to the standards, I also think it might be good in the "reference" section to provide examples of any scaffolding we DO have that follow these guidelines with maybe some short examples of how they fit?

Reply:
I don't know of any that currently do not adhere to the standards, but I can definitely add examples of ones that do - including the node scaffolding and ruby scaffolding.  The motive for this RFC was not so much that there are outstanding scaffoldings that don't adhere to these standards, but that there has been discussion around those that do.  I could definitely word that better!